spotbugs/src/main/java/edu/umd/cs/findbugs/detect/ViewCFG.java

```diff
@@ -0,0 +1,176 @@
+package edu.umd.cs.findbugs.detect;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Iterator;
+
+import org.apache.bcel.classfile.JavaClass;
+import org.apache.bcel.classfile.Method;
+import org.apache.bcel.generic.InstructionHandle;
+
+import edu.umd.cs.findbugs.BugReporter;
+import edu.umd.cs.findbugs.Detector;
+import edu.umd.cs.findbugs.ba.BasicBlock;
+import edu.umd.cs.findbugs.ba.CFG;
+import edu.umd.cs.findbugs.ba.CFGBuilderException;
+import edu.umd.cs.findbugs.ba.ClassContext;
+import edu.umd.cs.findbugs.ba.Edge;
+import static edu.umd.cs.findbugs.ba.EdgeTypes.*;
+
+public class ViewCFG implements Detector {
+
+    private final BugReporter bugReporter;
+    private Path tempDir;
+
+    public ViewCFG(BugReporter bugReporter) {
+        this.bugReporter = bugReporter;
+        try {
+            tempDir = Files.createTempDirectory("cfg-");
+        } catch (IOException e) {
+            bugReporter.logError("Could not create temporary directory", e);
+        }
+    }
+
+    @Override
+    public void visitClassContext(ClassContext classContext) {
+        if (tempDir == null) {
+            return;
+        }
+        //
+        JavaClass cls = classContext.getJavaClass();
+        String classDirName = (cls.getPackageName() != "") ? (cls.getPackageName() + "." + cls.getClassName()) : cls.getClassName();
```

Comment:
I've not tested in my local, however `$` is a special character for the file name in some file systems.
Please try to test with inner classes like `Outer$Inner`, then we may find some problems.

Reply:
All the file systems (Unix file systems, DOS/Windows file systems, even OpenVMS) I know of accept a $ sign in file or directory names. I tested it now on a Linux (ext4) and NTFS and I did not have any problems. Could you please provide me an example where it may cause problems?

Reply:
Ah, it's my bad. When I execute `touch Outer$Inner` I found there is no `Outer$Inner` file created, but I should execute `touch Outer\$Inner` instead. :sob: